driver/src/driver/stablex_driver.rs

```diff
@@ -85,11 +85,34 @@ impl<'a> StableXDriver<'a> {
             self.metrics
                 .auction_solution_verified(batch_to_solve, &verification_result);
 
-            let objective_value = verification_result?;
+            match verification_result {
+                Ok(objective_value) => {
+                    info!(
+                        "Verified solution with objective value: {}",
+                        objective_value
+                    );
+                    Some(objective_value)
+                }
+                Err(err) => match err {
+                    SolutionSubmissionError::Benign(reason) => {
+                        info!("Benign failure while verifying solution: {}", reason);
+                        None
+                    }
+                    SolutionSubmissionError::Unexpected(err) => {
+                        // Return from entire function with the unexpected error
+                        return Err(err);
+                    }
+                },
+            }
+        } else {
             info!(
-                "Verified solution with objective value: {}",
-                objective_value
+                "Not submitting trivial solution for batch {}",
+                batch_to_solve
             );
+            None
```

Comment:
Ubernit: this branch and the one in lines 128-131 are equivalent. I would probably just add the metric and return false here right away.

Reply:
But we also return None in line 99 when it's a Benign error. We would have to log and return there as well, which I thought complicates the control flow (I'm overall not super happy how this control flow turned out, it's a bit messy)

Reply:
Oh I missed that, you are absolutely right!